### PR TITLE
This fixes the Jaeger integration test and generally cleans it up

### PR DESCRIPTION
**Pull request by adriancole**
Before, the Jaeger integration test hand-rolled docker setup which
failed on my machine, distracting from other unrelated change. When
looking at it, I noticed it wasn't using the normal test libraries as well.
This cleans up everything and now works as expected.

### CODE REVIEW

File: exporters/trace/jaeger/src/test/java/io/opencensus/exporter/trace/jaeger/JaegerExporterHandlerIntegrationTest.java [hunk 1]
```diff
@@ -46,219 +40,167 @@
...
-    assumeThat("docker is installed and running", isDockerInstalledAndRunning(), is(true));
-
-    final Process jaeger =
-        Runtime.getRuntime()
-            .exec(
-                "docker run --rm "
-                    + "-e COLLECTOR_ZIPKIN_HTTP_PORT=9411 -p5775:5775/udp -p6831:6831/udp "
-                    + "-p6832:6832/udp -p5778:5778 -p16686:16686 -p14268:14268 -p9411:9411 "
-                    + "jaegertracing/all-in-one:1.2.0");
-    waitForJaegerToStart(format("http://%s:16686", JAEGER_HOST));
```
Comment (adriancole) [inline exporters/trace/jaeger/src/test/java/io/opencensus/exporter/trace/jaeger/JaegerExporterHandlerIntegrationTest.java]: indents invalidated a lot of this as well changing the assertions. Trust me.. it is ok :P

**Comment by adriancole** [comment]
had to play a little game to disable running docker test on JDK 1.7 (which is totally fine as other tests still run)

**Comment by adriancole** [comment]
ok good to go now

File: exporters/trace/jaeger/src/test/java/io/opencensus/exporter/trace/jaeger/JaegerExporterHandlerIntegrationTest.java (hunk 1 above)
Comment (sebright) [reply exporters/trace/jaeger/src/test/java/io/opencensus/exporter/trace/jaeger/JaegerExporterHandlerIntegrationTest.java]: I was able to view the changes with --color-words and --ignore-all-space.

**Comment by adriancole** [comment]
added a commit to skip when there's any problem initializing docker (as appveyor, the windows slave does not have docker)

**Comment by adriancole** [comment]
re-attempt to workaround appveyor's lack of docker

**Pull request merged**